sway-core/src/parse_tree/literal.rs

```diff
@@ -241,6 +253,33 @@ impl Literal {
     pub(crate) fn new_pointer_literal(offset_bytes: u64) -> Literal {
         Literal::U64(offset_bytes)
     }
+
+    #[allow(clippy::wildcard_in_or_patterns)]
+    pub(crate) fn handle_parse_int_error(
+        e: ParseIntError,
+        ty: TypeInfo,
+        span: Span,
+        path: Option<Arc<PathBuf>>,
+    ) -> CompileError {
+        match e.kind() {
+            IntErrorKind::PosOverflow => CompileError::IntegerTooLarge {
+                ty: ty.friendly_type_str(),
+                span: span::Span { span, path },
+            },
+            IntErrorKind::NegOverflow => CompileError::IntegerTooSmall {
+                ty: ty.friendly_type_str(),
+                span: span::Span { span, path },
+            },
+            IntErrorKind::InvalidDigit => CompileError::IntegerContainsInvalidDigit {
+                ty: ty.friendly_type_str(),
+                span: span::Span { span, path },
+            },
+            IntErrorKind::Zero | IntErrorKind::Empty | _ => CompileError::Internal(
```

Comment:
why do the wildcard if you're going to match the other two kinds?

In general, I'd say prefer to omit the wild card. That way, when you add variants to an enum, the compiler points out all the places it needs to be handled.

Reply:
I actually just moved this function inside `impl Literal` without changing it, but I can fix it anyways.

Reply:
Ah I think I actually wrote this line. Well, I stand by my comment 😄 live and learn

Reply:
Even though the wildcard is not necessary (see https://doc.rust-lang.org/std/num/enum.IntErrorKind.html), the compiler still complains when I remove it, and says that not all possible cases are being handled. I'm a little puzzled. Is it the `#[non_exhaustive]` annotation that causes this?